### PR TITLE
AArch64: Implement Table Evaluator

### PR DESCRIPTION
**Pull request by aarongraham9**
Implements Table Evaluator for AArch64

Co-authored-by: Aaron Graham <aaron.graham@unb.ca>
Co-authored-by: Md. Mahbubur Rahman <mrahma15@unb.ca>

Signed-off-by: Aaron Graham <aaron.graham@unb.ca>

### CODE REVIEW

**Comment by aarongraham9** [comment]
This PR: subsumes, takes in the feedback from and continues PR #3858.

Rebased off of the `master` branch and removed @knn-k's temporary implementation of the table evaluator from #4163.

This should be good for a review @0xdaryl.

**Comment by aarongraham9** [comment]
This needs a rebase, but otherwise, should be good for a review @0xdaryl.

File: compiler/aarch64/codegen/ControlFlowEvaluator.cpp [hunk 1]
```diff
@@ -23,6 +23,7 @@
 #include "codegen/ARM64Instruction.hpp"
 #include "codegen/CodeGenerator.hpp"
 #include "codegen/GenerateInstructions.hpp"
+#include "codegen/OMRCodeGeneratorUtils.hpp"
```
Comment (knn-k) [inline compiler/aarch64/codegen/ControlFlowEvaluator.cpp]: Include "codegen/CodeGeneratorUtils.hpp" (without OMR) instead.

File: compiler/aarch64/codegen/ControlFlowEvaluator.cpp [hunk 2]
```diff
@@ -476,8 +477,70 @@ OMR::ARM64::TreeEvaluator::lookupEvaluator(TR::Node *node, TR::CodeGenerator *cg
...
+         }
+      else
+         {
+         generateCompareImmInstruction(cg, node, selectorReg, numBranchTableEntries);
+         }
+
+      generateConditionalBranchInstruction(cg, TR::InstOpCode::b_cond, node, defaultChild->getBranchDestination()->getNode()->getLabel(), TR::CC_CS);
+      generateCompareImmInstruction(cg, node, t1Register, 1);
+      generateLogicalShiftLeftImmInstruction(cg, node, t1Register, selectorReg, 2);
+      generateCompareInstruction(cg, node, selectorReg, t1Register);
```
Comment (knn-k) [inline compiler/aarch64/codegen/ControlFlowEvaluator.cpp]: The results of these compare and shift instructions are never used.

**Comment by aarongraham9** [comment]
This PR (#4214) has been superseded by #4633. Closing.

**Pull request closed without merging**